engine/function/src/templates/Numeric.ftl

```diff
@@ -1779,51 +1861,105 @@ public class Numeric {
 
         return result;
     }
+    <#else>
+    public static long[] cumsum(${pt.vector} values) {
+        if (values == null) {
+            return null;
+        }
+
+        if (values.isEmpty()) {
+            return new long[0];
+        }
+
+        final int n = values.intSize("cumsum");
+        final long[] result = new long[n];
+
+        try ( final ${pt.vectorIterator} vi = values.iterator() ) {
+            final ${pt.primitive} v0 = vi.${pt.iteratorNext}();
+            result[0] = isNull(v0) ? NULL_LONG : v0;
+            int i = 1;
+
+            while (vi.hasNext()) {
+                final ${pt.primitive} v = vi.${pt.iteratorNext}();
```

Comment:
With a separate function for `float` inputs, we **could** short-circuit future calculations once poisoned by `NaN`.  Might not save much, since we have to write a value for every index but could leverage `Arrays.fill()` for efficient filling.

To be clear, I'm not convinced this is worth doing but we are doing it for `product` currently.